### ISSUE TITLE
InvalidMixinException thrown when MovingWorld mod is included (loading MovingWorld's older Mixin library before Sponge's)

### ISSUE THREAD

**Issue by ghost** [SpongePowered/SpongeForge#1631]
**I am currently running...**
- SpongeForge version:  2436
- Forge version: 2397
- Java version:  8u131
- Operating System: windows 10 x64
- Plugins/Mods: 
universeguard
QUtilities

**What is the expected result?**
my dedicated server should start
**What is the current result?**
it didn't

[fml-junk-earlystartup.log.txt](https://github.com/SpongePowered/SpongeForge/files/1128360/fml-junk-earlystartup.log.txt)

**Comment by Mumfrey** [comment]
This is happening because Mixin in MovingWorld is loading before Mixin in Sponge and @gabizou forgot to change the `minVersion` in the config file which *would* have made this crash more obvious.

Force sponge to load before movingWorld by renaming one of the mods.

**Comment by ghost** [comment]
wa wait, how??

**Comment by ghost** [comment]
btw, i tried deleting movingworld and davinci's vessels, but it still doesn't work. am i wrong?

**Comment by Mumfrey** [comment]
Change the filename of spongeForge to like `aaaSpongeForge.jar` so that it loads first.

See this line in your log:

```
[14:15:12] [main/INFO] [mixin/]: SpongePowered MIXIN Subsystem Version=0.6.9 Source=file:/C:/Users/ahmet.DESKTOP-8HLRTF0/Desktop/magenjianserver%20-%20Kopya/mods/movingworld-1.12-2.337-full.jar Env=SERVER
```

Unless that says `Source=file:// ... SpongeForge ...` then it's loading Mixin from somewhere else and thus is gonna break.

**Comment by ghost** [comment]
oh.... i got it. hope it work, thanks.

**Comment by ghost** [comment]
oops, i made spongeforga.jar to aaaspongeforge.jar as you said, but it didn't work. same thing is happening again. :( 
[fml-server-latest.log.txt](https://github.com/SpongePowered/SpongeForge/files/1125497/fml-server-latest.log.txt)
used sponge 2436

**Comment by ghost** [comment]
any other idea? =/

**Comment by Mumfrey** [comment]
You have successfully fixed the original issue. You are now seeing a different exception caused most likely by another mod.

**Comment by ghost** [comment]
yeah it doesn't happen with no mod. i think i have to find which mods are causing :/ really, isn't there another way to solve this problem?

**Comment by phit** [comment]
https://forums.spongepowered.org/t/spongebootstrap-fixes-ordering-issue-with-other-coremods/12275

**Comment by ghost** [comment]
well ah, i deleted bunch of other mods and got it working BUT server stops instantly just after the world generation (after %100) and here are the logs that i couldn't understand ANYthing cus nothing tells nothing about the crash (no crash reports)
[latest.log.txt](https://github.com/SpongePowered/SpongeForge/files/1128174/latest.log.txt)
[fml-server-latest.log.txt](https://github.com/SpongePowered/SpongeForge/files/1128175/fml-server-latest.log.txt)
[fml-junk-earlystartup.log.txt](https://github.com/SpongePowered/SpongeForge/files/1128176/fml-junk-earlystartup.log.txt)
and please tell me if i should open new issue...
edit: i think i understood what causes to this problem, the Ruins mod.

`[16:33:32] [Server thread/ERROR] [FML]: Exception caught during firing event net.minecraftforge.event.world.WorldEvent$Save@75ca7982:
java.lang.NullPointerException: null
	at net.minecraft.world.World.handler$onGetWorldInfo$zpc000(World.java:4642) ~[ams.class:?]
	at net.minecraft.world.World.func_72912_H(World.java) ~[ams.class:?]
	at atomicstryker.ruins.common.RuinsMod.eventWorldSave(RuinsMod.java:118) ~[RuinsMod.class:?]
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_402_RuinsMod_eventWorldSave_Save.invoke(.dynamic) ~[?:?]
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90) ~[ASMEventHandler.class:?]
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:634) [EventBus.class:?]
	at org.spongepowered.mod.event.SpongeForgeEventFactory.callWorldSaveEvent(SpongeForgeEventFactory.java:1203) [SpongeForgeEventFactory.class:1.12-2387-7.0.0-BETA-2436]
	at org.spongepowered.mod.event.SpongeForgeEventFactory.callForgeEvent(SpongeForgeEventFactory.java:481) [SpongeForgeEventFactory.class:1.12-2387-7.0.0-BETA-2436]
	at org.spongepowered.mod.event.SpongeModEventManager.post(SpongeModEventManager.java:291) [SpongeModEventManager.class:1.12-2387-7.0.0-BETA-2436]
	at org.spongepowered.mod.event.SpongeModEventManager.post(SpongeModEventManager.java:341) [SpongeModEventManager.class:1.12-2387-7.0.0-BETA-2436]
	at org.spongepowered.mod.event.SpongeModEventManager.post(SpongeModEventManager.java:329) [SpongeModEventManager.class:1.12-2387-7.0.0-BETA-2436]
	at net.minecraft.world.WorldServer.handler$onSaveAllChunks$zkc000(WorldServer.java:2368) [om.class:?]
	at net.minecraft.world.WorldServer.func_73044_a(WorldServer.java:1011) [om.class:?]
	at org.spongepowered.common.world.WorldManager.saveWorld(WorldManager.java:518) [WorldManager.class:1.12-2387-7.0.0-BETA-2436]
	at org.spongepowered.common.world.WorldManager.unloadWorld(WorldManager.java:497) [WorldManager.class:1.12-2387-7.0.0-BETA-2436]
	at net.minecraftforge.common.DimensionManager.setWorld(DimensionManager.java:653) [DimensionManager.class:?]
	at net.minecraft.server.MinecraftServer.func_71260_j(MinecraftServer.java:459) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:576) [MinecraftServer.class:?]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_131]`
and i think we will need help from @atomicstryker :D

**Comment by bookerthegeek** [comment]
@kenijey 
Try this when pasting short logs...

    ```
    Your log on the next line
    As many lines as you like
    This will make it easier to read if it is more then a single line. Another option is to use pastebin or ghist. This greatly improves readability
    ```

**Comment by ghost** [comment]
very sadly, it happens without ruins mod too... sorry.
this time QUtilities mod caused. u can read in this log
[fml-junk-earlystartup.log.txt](https://github.com/SpongePowered/SpongeForge/files/1128360/fml-junk-earlystartup.log.txt)
edit: and yes, worked after deleting QUtilities mod.

**Comment by AtomicStryker** [comment]
Crashes on World.getWorldInfo().getWorldName()

**Comment by dualspiral** [comment]
Let's keep sight of what this ticket is for - you're now adding errors that are nothing to do with the original issue. This issue should _only_ be for the `InvalidMixinError` issue (for which the problem has been identified, it's just waiting on the fix).

@kenijey, your other errors (which, as @Mumfrey said, are different to the original issue) should be put in as seperate tickets (but please search for them in the issue tracker first to make sure they haven't been reported yet).

**Comment by ghost** [comment]
theres only 1 issue we are having right now, i can edit this issue. let me do it please.

**Comment by ghost** [comment]
and i can open new issue based on the original issue that i had at the beginning if anybody wants. (renaming the sponge.jar to a0sponge.jar)

**Comment by dualspiral** [comment]
@kenijey Please stop a moment. I am going to revert _this_ ticket's title to be what it originally was (because it has the commentary on it already) and ask you to make a _new_ one for your QUtilities problem.

**Please note that GitHub is not a support forum**, and while we really do care that you are having issues with your server, GitHub issues are for tracking code defects, and not for tracking issues as specific server is having. While you were given a _workaround_ for the `InvalidMixinException`, it is not a _fix_. This issue **must not** be edited to be an entirely different issue that you encounter - because we then lose sight of what the _original issue_ was, which we still need to fix. This issue is _not_ for continuously tracking problems with your server, and should not be treated as such.

The reason I asked you to open a separate ticket is so that we have something to remind us of each issue you are having, so that we know what we need to do. One problem, one separate issue please.

**Comment by Zidane** [comment]
Closing this as the root issue has been resolved.
